amp-pwa-reader/src/js/Article.js

```diff
@@ -41,6 +42,61 @@ class Article {
 
   }
 
+// see https://github.com/ampproject/amphtml/blob/master/spec/amp-shadow-doc.md
+// and https://jakearchibald.com/2016/fun-hacks-faster-content/
+// Recursively read and write chunks of data from the streaming API
+  stream() {
+    var shadowDoc = this.ampDoc;    // let's get this into the closure, and thus accessible to callback
+    var article = this;             // this too
+
+    fetch(this.proxyUrl).then(async response => {
+      let reader = response.body.getReader();
+      let decoder = new TextDecoder();
+
+      while (true) {
+        let chunk = await reader.read();
+
+        if (chunk.done) {
+          shadowDoc.writer.close();
+          break;
+        }
+
+        let html = decoder.decode(
```

Comment:
nit: can we call this something like partialHtml or htmlPart to make it clear that this is a chunk of html?

Reply:
@dvoytenko called it `text` [here](https://github.com/ampproject/amphtml/blob/master/examples/pwa/pwa.js).  This made sense to me! I simply used `html` because it told me the type of thing the variable contained.

`html` implies to you that it's the whole page html?  Should I then call it, say, `htmlChunk` for consistency? To imply that it's the decoded html?

Reply:
For consistency is a chunk or htmlChunk or decodedChunk - all chunks are good.

Reply:
@pbakaus are we good with `html`, or do you prefer, say, `htmlChunk`?

Reply:
maybe `htmlChunk`, sgtm?